File: AUGUST-2025/01-08-2025/p118.cpp
class Solution {
public:
    vector<vector<int>> generate(int n) {
        vector<vector<int>> result;
        result.push_back({1});
        if(n==1) return result;

        result.push_back({1,1});

        for(int i=3; i<=n; i++){
            vector<int> temp;
            temp.push_back(1);
            int y=1;
            while(y<i-1){
                temp.push_back(result.back()[y-1]+result.back()[y]);
                y++;
            }
            temp.push_back(1);
            result.push_back(temp);
            
        }

        return result;
    }
};

File: AUGUST-2025/01-08-2025/p118.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        
        
        if numRows==1:
            return [[1]]

        if numRows==2:
            return [[1],[1,1]]

        res=[[1]]

        for i in range(numRows-1):
            curr=[]
            curr.append(1)
            for j in range(i):
                curr.append(sum(res[-1][j:j+2]))
            curr.append(1)
            res.append(curr)

        return res